lib/ts/BufferWriterFormat.cc

```diff
@@ -211,24 +211,25 @@ namespace bw_fmt
           *dst++ = spec._fill;
         break;
       case BWFSpec::Align::CENTER:
-        d2 = (delta + 1) / 2;
-        if (d2 > 1) {
-          dst = base + d2; // move existing content to here.
-          if (dst < limit) {
-            last = dst + size; // amount of data to move.
-            if (last > limit)
-              last = limit;
-            std::memmove(dst, base, last - dst);
-          }
-          dst  = base + size + d2;
-          last = base + delta / 2;
+        d2 = (delta + 1) / 2; // always > 0 because min > extent
+        // Move the original content right to make space to fill on the left.
+        dst = base + d2; // move existing content to here.
+        if (dst < limit) {
+          last = dst + size; // amount of data to move.
           if (last > limit)
             last = limit;
-          while (dst < last)
-            *dst++ = spec._fill;
+          std::memmove(dst, base, last - dst);
         }
+        // Left fill.
         dst  = base;
         last = base + d2;
+        if (last > limit)
+          last = limit;
```

Comment:
Should this ever happen? If `last = limit` then from `base` to `limit` (entire) buffer will be written with `spec.fill`.

Reply:
That is by design. Suppose `min == 10` and the output was 6 long. Then there should be 2 fill on the left. If there was only 1 byte remaining in the buffer, then it should get the fill character (e.g. fill from base to limit) and nothing else, as the actual output was pushed beyond the end of the buffer by the alignment. As noted another response, this should behave as if the buffer were infinite and then clipped at the very end.